EIPS/eip-draft_bls_keygen.md

```diff
@@ -0,0 +1,280 @@
+---
+eip: <to be assigned>
+title: BLS12-381 Key Generation
+author: Carl Beekhuizen (carl@ethereum.org)
+discussions-to: https://github.com/ethereum/EIPs/issues/2337
+status: Draft
+type: Standards Track
+category: ERC
+created: 2019-09-30
+---
+
+## Simple Summary
+
+This EIP is a method based on a tree structure for deriving BLS private keys from a single source of entropy while providing a post-quantum cryptographic fallback for each key.
+
+## Abstract
+
+This standard is a method for deriving a tree-hierarchy of BLS12-381 keys based on an entropy seed. Starting with the aforementioned seed, a tree of keys is built out using only the parent node's private key and the index of the desired child. This allows for a practically limitless number of keys to be derived for many different purposes while only requiring knowledge of a single ancestor key in the tree. This allows for keys, or families thereof, to be provisioned for different purposes by further standards.
+
+In addition to the above, this method of deriving keys provides an emergency backup signature scheme that is resistant to quantum computers for in the event that BLS12-381 is ever deemed insecure.
+
+## A note on purpose
+
+This specification is designed not only to be an Ethereum 2.0 standard, but one that is adopted by the wider community who have adopted the BLS12-381 signature standard. It is therefore important also to consider the needs of the wider industry along with those specific to Ethereum. As a part of these considerations, it is the intention of the author that this standard eventually migrate to a more neutral repository in the future.
+
+## Motivation
+
+### Deficiencies of the existing mechanism
+
+The new curve (BLS12-381) used for signatures within Ethereum 2.0 (alongside many other projects) mandates a new key derivation scheme. The most commonly used scheme for key derivation at the within Ethereum 1.x is [BIP32](https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki) (also known as HD derivation) which deems keys greater than the curve order invalid. Based on the order of the private key subgroup of BLS12-381 and the size of the entropy utilised, more than 54% of keys generated by BIP32 would be invalid. (secp256k1 keys derived by BIP32 are invalid with probability less than 1 in 2<sup>-127</sup>.)
```

Comment:
Should this say `have a probability of 1 in 2^127` to be valid? As it currently stands it sounds like the odds are in favor of secp256k1 keys.